python/servo/command_base.py

```diff
@@ -422,13 +422,15 @@ def ensure_bootstrapped(self, targets=[]):
         rustc_binary_exists = path.exists(rustc_path)
 
         base_target_path = path.join(rust_root, "rustc", "lib", "rustlib")
-        target_paths = [path.join(base_target_path, t) for t in targets]
-        all_targets_exist = all([path.exists(p) for p in target_paths])
+        target_exists = False
+        if target is not None:
+            target_path = path.join(base_target_path, target)
+            target_exists = path.exists(target_path)
 
         if (not self.config['tools']['system-rust'] and
-                (not rustc_binary_exists or not all_targets_exist)):
+                (not rustc_binary_exists or not target_exists)):
```

Comment:
I find this whole if chain hard to read as-is due to all the nots. Let's change it to:
```
if not (self.config['tools']['system-rust'] or (rustc_binary_exists and target_exists)):
```
(Modulo formatting)